test/expect-tests/dune_sexp/dune_sexp_tests.ml

```diff
@@ -0,0 +1,80 @@
+open Stdune
+
+let () = Dune_tests_common.init ()
+
+(* Testing the parsing of byte values *)
+let parse_bytes value =
+  Dune_sexp.Ast.atom_or_quoted_string Loc.none value
+  |> Dune_sexp.Decoder.parse Dune_sexp.Decoder.bytes_unit Univ_map.empty
+;;
+
+let test_bytes value = parse_bytes value |> Printf.printf "%#Ld\n"
+
+(* Hack to insert underscores for hex values. Digits must only be 0-9 *)
+let test_bytes_hex value =
+  parse_bytes value
+  |> sprintf "%Lx"
+  |> Int.of_string
+  |> function
+  | Some x -> x |> Printf.sprintf "0x%#d\n" |> print_endline
+  | None -> print_endline "hex value must not have letters"
+;;
+
+(* Test parsing of integers. *)
+
+let%expect_test "parsing no suffix" =
+  test_bytes "100";
+  [%expect.unreachable]
+[@@expect.uncaught_exn
+  {|
+  ( "File \"<none>\", line 1, characters 0-0:\
+   \nError: missing suffix, use one of B, kB, KiB, MB, MiB, GB, GiB, TB, TiB\
+   \n") |}]
+;;
+
+(* Test all suffixes. We print binary units in hex to better see output. *)
+
+let%expect_test "parsing B suffix" =
+  test_bytes "1B";
+  [%expect {| 1 |}]
+;;
+
+let%expect_test "parsing kB suffix" =
+  test_bytes "1kB";
+  [%expect {| 1_000 |}]
+;;
+
+let%expect_test "parsing KiB suffix" =
+  test_bytes_hex "1KiB";
+  [%expect {| 0x400 |}]
+;;
+
+let%expect_test "parsing MB suffix" =
+  test_bytes "1MB";
+  [%expect {| 1_000_000 |}]
+;;
+
+let%expect_test "parsing MiB suffix" =
+  test_bytes_hex "1MiB";
+  [%expect {| 0x100_000 |}]
+;;
+
+let%expect_test "parsing GB suffix" =
+  test_bytes "1GB";
+  [%expect {| 1_000_000_000 |}]
+;;
+
+let%expect_test "parsing GiB suffix" =
+  test_bytes_hex "1GiB";
+  [%expect {| 0x40_000_000 |}]
```

Comment:
Same here: maybe output both hexadecimal and decimal numbers, so that I can at least quickly convince myself that the number of digits in the decimal number is as expected.

Reply:
I've replaced this with an assertion to a long power.